pkg/timeutil/timespan.go

```diff
@@ -0,0 +1,186 @@
+package timeutil
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	badDay = time.Weekday(-1)
+
+	shortDayNames = map[string]time.Weekday{
+		"Sun": time.Sunday,
+		"Mon": time.Monday,
+		"Tue": time.Tuesday,
+		"Wed": time.Wednesday,
+		"Thu": time.Thursday,
+		"Fri": time.Friday,
+		"Sat": time.Saturday,
+	}
+)
+
+func d2s(d time.Weekday) string {
+	for k, v := range shortDayNames {
+		if d == v {
+			return k
+		}
+	}
```

Comment:
good idea. unfortunately the maps of day constants to day names and vice-versa are not exported from `time`